qiskit_aer/quantum_info/states/aer_statevector.py

```diff
@@ -0,0 +1,273 @@
+        return AerStatevector(data, _aer_state=aer_state, **self._configs)
+
+    def __eq__(self, other):
+        eq_quantumqtate = isinstance(other, Statevector) and self.dims() == other.dims()
+        eq_statevector = np.allclose(self._data, other._data, rtol=self.rtol, atol=self.atol)
+        return eq_quantumqtate and eq_statevector
+
+    def evolve(self, other, qargs=None):
+        """Evolve a quantum state by the operator.
+
+        Args:
+            other (Operator): The operator to evolve by.
+            qargs (list): a list of Statevector subsystem positions to apply
+                           the operator on.
+
+        Returns:
+            Statevector: the output quantum state.
+
+        Raises:
+            AerError: if the operator dimension does not match the
+                         specified Statevector subsystem dimensions.
+        """
+        if qargs is None:
+            qargs = getattr(other, "qargs", None)
+
+        if isinstance(other, (QuantumCircuit, Instruction)):
+            data, aer_state = AerStatevector._from_instruction(other, self._data, **self._configs)
+            return AerStatevector(data, _aer_state=aer_state, **self._configs)
+
+        ret = copy.copy(self)
+
+        # Evolution by an Operator
+        if not isinstance(other, Operator):
+            other = Operator(other)
+
+        # check dimension
+        if self.dims(qargs) != other.input_dims():
+            raise AerError(
+                "Operator input dimensions are not equal to statevector subsystem dimensions."
+            )
+        return Statevector._evolve_operator(ret, other, qargs=qargs)
+
+    def sample_memory(self, shots, qargs=None):
+        """Sample a list of qubit measurement outcomes in the computational basis.
+
+        Args:
+            shots (int): number of samples to generate.
+            qargs (None or list): subsystems to sample measurements for,
+                                if None sample measurement of all
+                                subsystems (Default: None).
+
+        Returns:
+            np.array: list of sampled counts if the order sampled.
+
+        Additional Information:
+
+            This function *samples* measurement outcomes using the measure
+            :meth:`probabilities` for the current state and `qargs`. It does
+            not actually implement the measurement so the current state is
+            not modified.
+        """
+        if qargs is None:
+            qubits = np.arange(self._aer_state.num_qubits)
+        else:
+            qubits = np.array(qargs)
+        self._aer_state.close()
+        self._aer_state = AerState(**self._aer_state.configuration())
+        self._aer_state.initialize(self._data, copy=False)
+        samples = self._aer_state.sample_memory(qubits, shots)
+        self._data = self._aer_state.move_to_ndarray()
+        return samples
+
+    @classmethod
+    def _from_ndarray(cls, init_data, **configs):
```

Comment:
This (and the others of its kind) is logically a `staticmethod`, not a `classmethod` (since they don't use `cls`).

Reply:
Right. I copied most of methods from `Statevector`, first. Interestingly, it uses `@classmethod` :-).